clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackServerGroup.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 Target Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.model
+
+import com.netflix.spinnaker.clouddriver.model.HealthState
+import com.netflix.spinnaker.clouddriver.model.Instance
+import com.netflix.spinnaker.clouddriver.model.ServerGroup
+import com.netflix.spinnaker.clouddriver.model.ServerGroup.InstanceCounts
+import com.netflix.spinnaker.clouddriver.model.ServerGroup.ImagesSummary
+import com.netflix.spinnaker.clouddriver.model.ServerGroup.ImageSummary
+import com.netflix.spinnaker.clouddriver.model.ServerGroup.Capacity
+import com.netflix.spinnaker.clouddriver.openstack.OpenstackCloudProvider
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.builder.Builder
+
+@Builder
+@EqualsAndHashCode
+class OpenstackServerGroup implements ServerGroup, Serializable {
+  String account
+  String name
+  String region
+  Set<String> zones
+  Set<OpenstackInstance> instances
+  Set health
+  Map<String, Object> image // Represented as map instead of OpenstackImage for convenience.
+  Map<String, Object> launchConfig
+  Map<String, Object> asg = [:]
+  Set<String> loadBalancers
+  Map<String, Object> buildInfo
+  Boolean disabled
+  String type = OpenstackCloudProvider.ID
+
+  @Override
+  Boolean isDisabled() {
+    disabled
+  }
+
+  @Override
+  Long getCreatedTime() {
+    asg ? asg.createdTime as Long : null
+  }
+
+  @Override
+  Set<String> getSecurityGroups() {
+    (launchConfig && launchConfig.containsKey('securityGroups')) ? (Set<String>) launchConfig.securityGroups : []
+  }
+
+  @Override
+  InstanceCounts getInstanceCounts() {
+    instances ?
+      new InstanceCounts(total: instances.size(),
+        up: filterInstancesByHealthState(instances, HealthState.Up)?.size() ?: 0,
+        down: filterInstancesByHealthState(instances, HealthState.Down)?.size() ?: 0,
+        unknown: filterInstancesByHealthState(instances, HealthState.Unknown)?.size() ?: 0,
+        starting: filterInstancesByHealthState(instances, HealthState.Starting)?.size() ?: 0,
+        outOfService: filterInstancesByHealthState(instances, HealthState.OutOfService)?.size() ?: 0)
+      : null
```

Comment:
Is this the right behavior? It feel like it should return 0 for all instance counts rather than `null` if there are no instances.